### PR TITLE
Update doc

### PR DESCRIPTION
**Pull request by weiqingy**
## What changes were proposed in this pull request?
Update doc.

### CODE REVIEW

**Comment by weiqingy** [comment]
This PR is to add doc for the code changes of supporting multiple secure HBase clusters. @jerryshao @mridulm Could you please help to review this? Thanks.

**[FYI] Some testing result updates**:
The testing worked well in both Yarn client mode and Yarn cluster mode. The change has been verified in a single secure cluster. The setup of secure multiple clusters has some issue, QE team is working on it. After fixing the issues, will do testing in multiple secure clusters. 

Cluster 1. Ran LRJobAccessing2Clusters example in Yarn cluster mode: succeed. Token refresh thread worked well. http://172.22.71.30:8088/cluster/app/application_1491095099401_0029

Cluster 2. Ran LRJobAccessing2Clusters example in Yarn client mode: succeed. Token refresh thread worked well. http://172.22.71.29:8088/cluster/app/application_1491095259289_0040

The configurations in spark-default.conf:

```
spark.yarn.security.credentials.hbase.enabled false
spark.hbase.connector.security.credentials  ambari-qa-c1@EXAMPLE.COM
spark.hbase.connector.security.keytab  /etc/security/keytabs/smokeuser.headless.keytab
```

**Comment by weiqingy** [comment]
cc: @bikassaha

File: README.md [hunk 1]
```diff
@@ -133,11 +133,24 @@ Users can include the package as the dependency in your SBT file as well. The fo
 
 ## Running in secure cluster
 
+There are two ways to run your applications in secure clusters. One is to use Spark internal HBaseCredentialProvider, the other is to
+use SHCCredentialsManager.
+
+Spark internal HBaseCredentialProvider only supports use cases which access a single secure HBase cluster. If your applications need to
+access multiple secure HBase clusters, please use SHCCredentialsManager instead. Also, HBaseCredentialProvider can not retrieve or renew
+HBase tokens, so it can not support long running applications running beyond the lifetimes of HBase tokens. Users can set _spark.yarn.security.credentials.hbase.enabled_
+to false/true to disable/enable HBaseCredentialProvider, which is enabled by default.
```
Comment (jerryshao) [inline README.md]: I don't think we need to mention `HBaseCredentialProvider` here, first it should not be related, second `HBaseCredentialProvider` supports re-acquiring tokens.

**Comment by weiqingy** [comment]
Will update the doc soon.

File: README.md [hunk 2]
```diff
@@ -160,6 +160,24 @@ or
     
 Include the hbase-site.xml under SPARK_CONF_DIR (/etc/spark/conf) on the host where the spark job is submitted from, by creating a symbolic link towards your main hbase-site.xml (in order to be synchronous with your platform updates).
 
+### Using SHCCredentialsManager
+
+Spark only supports use cases which access a single secure HBase cluster. If your applications need to access multiple secure HBase clusters, users need to use SHCCredentialsManager instead.
+SHCCredentialsManager supports a single secure HBase cluster as well as multiple secure HBase clusters. 
+
+For normal Spark applications which complete running before HBase tokens expire, the only thing users need to do is to set _spark.yarn.security.credentials.hbase.enabled_ to false.
+_**Note**: actually it's fine to do nothing here as both Spark and SHCCredentialsManager support normal applications_.
```
Comment (jerryshao) [inline README.md]: If it is not required, then I would suggest to remove this two sentences. I think `HBaseCredentialProvider` related things is unrelated to shc. We could turn it on by default, it should be OK without issue.

File: README.md [hunk 3]
```diff
@@ -136,8 +136,8 @@ Users can include the package as the dependency in your SBT file as well. The fo
 For running in a Kerberos enabled cluster, the user has to include HBase related jars into the classpath as the HBase token
 retrieval and renewal is done by Spark, and is independent of the connector. In other words, the user needs to initiate the
 environment in the normal way, either through kinit or by providing principal/keytab.  The following examples show how to run
-in a secure cluster with both yarn-client and yarn-cluster mode. Note that SPARK_CLASSPATH has to be set for both modes, and
-the example jar is just a placeholder for Spark.
+in a secure cluster with both yarn-client and yarn-cluster mode. **Note that** if your Spark includes the patch of _["SPARK-20059: Use the correct classloader for HBaseCredentialProvider"](https://github.com/apache/spark/pull/17388)_
```
Comment (bikassaha) [inline README.md]: Lets change the wording the other way round to - If your Spark does not contain SPARK-20059 then you need to set SPARK_CLASSPATH. That way it is the exception and not the norm.

File: README.md (hunk 2 above)
Comment (bikassaha) [inline README.md]: We should either ask to disable the config or say nothing. Disabling the config and then saying its ok to not disable is confusing.

File: README.md [hunk 4]
```diff
@@ -160,6 +160,24 @@ or
     
 Include the hbase-site.xml under SPARK_CONF_DIR (/etc/spark/conf) on the host where the spark job is submitted from, by creating a symbolic link towards your main hbase-site.xml (in order to be synchronous with your platform updates).
 
+### Using SHCCredentialsManager
+
+Spark only supports use cases which access a single secure HBase cluster. If your applications need to access multiple secure HBase clusters, users need to use SHCCredentialsManager instead.
+SHCCredentialsManager supports a single secure HBase cluster as well as multiple secure HBase clusters.
```
Comment (bikassaha) [inline README.md]: Is there a config to turn SHC cred manager on?

File: README.md [hunk 5]
```diff
@@ -160,6 +160,24 @@ or
...
+_**Note**: actually it's fine to do nothing here as both Spark and SHCCredentialsManager support normal applications_.
+
+For long running applications, users need to config principal and keytab as below before running their applications.
+
+    spark.yarn.principal   ambari-qa-c1@EXAMPLE.COM
+    spark.yarn.keytab      /etc/security/keytabs/smokeuser.headless.keytab
+
+or
+
+    spark.hbase.connector.security.credentials  ambari-qa-c1@EXAMPLE.COM
```
Comment (bikassaha) [inline README.md]: If SHC cred manager is enabled and default hbase cred manager is disabled and keytab is not provided will it work?

File: README.md (hunk 3 above)
Comment (weiqingy) [inline README.md]: Sure.

File: README.md (hunk 4 above)
Comment (weiqingy) [inline README.md]: Yes. `spark.hbase.connector.security.credentials.enabled`.

File: README.md (hunk 2 above)
Comment (weiqingy) [inline README.md]: Deleted it.

Comment (weiqingy) [inline README.md]: Deleted it.

File: README.md [hunk 6]
```diff
@@ -160,6 +160,21 @@ or
     
 Include the hbase-site.xml under SPARK_CONF_DIR (/etc/spark/conf) on the host where the spark job is submitted from, by creating a symbolic link towards your main hbase-site.xml (in order to be synchronous with your platform updates).
 
+### Using SHCCredentialsManager
+
+Spark only supports use cases which access a single secure HBase cluster. If your applications need to access multiple secure HBase clusters, users need to use SHCCredentialsManager instead.
+SHCCredentialsManager supports a single secure HBase cluster as well as multiple secure HBase clusters. It is enabled by default, but users can set _spark.hbase.connector.security.credentials.enabled_ to false to disable it.
```
Comment (bikassaha) [inline README.md]: How can SHC cred manager be enabled by default when its applicable on to yarn-client mode? For the normal case of running a job using shc that access only 1 hbase cluster should we slow down the query by fetching the token when the query runs vs using the token that the default hbase manager has already obtained before driver started?

File: README.md (hunk 5 above)
Comment (weiqingy) [inline README.md]: Emm... I think I missed something in Jerry's latest PR https://github.com/hortonworks-spark/shc/pull/122. In the PR, if credentialsManagerEnabled is true/enabled, users have to set  keytab and principle, no matter if it is common jobs or long running jobs. @jerryshao

**Comment by weiqingy** [comment]
Submitted a new [PR#131](https://github.com/hortonworks-spark/shc/pull/131) for the doc and making the default value of credentialsManagerEnabled false. So close this.

**Pull request closed without merging**